packages/gateway/src/features/registration/root-resolvers.ts

```diff
@@ -278,20 +281,45 @@ export const resolvers: GQLResolver = {
       }
     },
     async markBirthAsValidated(_, { id, details }, authHeader) {
-      if (hasScope(authHeader, 'validate')) {
-        await markEventAsValidated(id, authHeader, EVENT_TYPE.BIRTH, details)
+      const hasAssignedToThisUser = await checkUserAssignment(id, authHeader)
+      if (!hasAssignedToThisUser) {
+        throw new ApolloError('ASSIGNMENT_PROBLEM', 'UNASSIGNED')
```

Comment:
Could we make our own class for this like `AssignmentError`? Notice also that the first parameter of `ApolloError` seems to be a human readable error message.

https://www.apollographql.com/docs/apollo-server/data/errors/#custom-errors

Reply:
I have added changes in [below commit](https://github.com/opencrvs/opencrvs-core/pull/3144/commits/bc44cd4f45c5b99fffec47422814918de380b105) , could you please have a look?

Reply:
@rikukissa I have introduced a custom error class [here](https://github.com/opencrvs/opencrvs-core/pull/3144/commits/bf2fb38dd53feca5ca54f6a537b17663317426bc)